### PR TITLE
Verbose function create wait for logs before exiting

### PR DESCRIPTION
**Pull request by glyn**
Fixes https://github.com/projectriff/riff/issues/1215

### CODE REVIEW

**Comment by glyn** [comment]
> Generally looks good, but feels like we are abusing the errChan. Maybe the channel just needs to be renamed to something more descriptive of its role.

Yes, I was conscious of that problem, but I'll give it more thought. Thanks for raising it!

**Comment by scothis** [comment]
One of the `function create` commands stalled. I can't say that it's because of this change, but it's certainly in the same area of the codebase.

https://travis-ci.com/projectriff/riff/jobs/190046137#L3722

**Comment by glyn** [comment]
> One of the `function create` commands stalled. I can't say that it's because of this change, but it's certainly in the same area of the codebase.
> 
> https://travis-ci.com/projectriff/riff/jobs/190046137#L3722

Hmmm. I can't immediately see why my code change would have caused this. I wonder if it's our old friend https://github.com/projectriff/riff/issues/1121 recurring?

File: pkg/core/function.go [hunk 1]
```diff
@@ -242,6 +248,7 @@ func (c *client) makeBuildSourceSpec(options CreateFunctionOptions) *build.Sourc
 }
 
 func (c *client) displayFunctionCreationProgress(serviceNamespace string, serviceName string, logWriter io.Writer, stopChan <-chan struct{}, errChan chan<- error) {
+	defer close(errChan)
```
Comment (glyn) [inline pkg/core/function.go]: This avoids a deadlock in the `--local-path` case where `displayFunctionCreationProgress` was returning **without** closing `errChan`.

**Comment by glyn** [comment]
> > One of the `function create` commands stalled. I can't say that it's because of this change, but it's certainly in the same area of the codebase.
> > https://travis-ci.com/projectriff/riff/jobs/190046137#L3722
> 
> Hmmm. I can't immediately see why my code change would have caused this. I wonder if it's our old friend #1121 recurring?

My code changes were the cause. A `--local-path` build had a return path that did not close `loggingTerminationChan`.

**Comment by scothis** [comment]
@glyn can you merge/rebase master, there's a AZP patch that should help unblock things

**Comment by scothis** [comment]
These test failures are real

**Comment by glyn** [comment]
> These test failures are real

Not sure how to interpret this. Are the failures real at master too or are they specific to this PR? Re-merged master to try to see the wood for the trees...

**Comment by glyn** [comment]
Failed again with:
```
Function did not produce expected result:
   expected: RIFF
   actual: curl http://10.30.0.245:32202/ -H 'Host: fats-local-uppercase-node.riff-1771-2.example.com' -H 'Content-Type: text/plain' -d riff -v
The command "ci/fats.sh lite" exited with 1.
```
A test PR (with just a README change) based on HEAD of the master branch passes all the checks, so it seems the above failure is caused by this PR's code changes.

File: pkg/core/function.go [hunk 2]
```diff
@@ -185,21 +185,27 @@ func (c *client) CreateFunction(buildpackBuilder Builder, options CreateFunction
...
+				}
 			}
-			err := c.waitForSuccessOrFailure(ns, s.Name, 1, stopChan, errChan, options.Verbose)
 			if err != nil {
 				return nil, nil, nil, err
 			}
 
 			// fetch latest Service and Revision, ignore errors
-			s1, err := c.serving.ServingV1alpha1().Services(s.Namespace).Get(s.Name, metav1.GetOptions{})
+			s1, err := c.serving.ServingV1alpha1().Services(s.Namespace).Get(s.Name, v1.GetOptions{})
```
Comment (scothis) [inline pkg/core/function.go]: please revert to `metav1`, the name `v1` is too generic and leads to conflicts with other `v1` packages.

Comment (glyn) [reply pkg/core/function.go]: Fixed.

**Pull request merged**